File: src/leetcode/array/TwoSum.java
package leetcode.array;

import utility.array.ArraysHelper;

import java.util.HashMap;

public class TwoSum {


    public static void main(String[] args) {
        int[] arr = new int[]{2, 7, 11, 15};
        int target = 9;

        int[] arr1 = new int[]{3, 2, 4};
        int target1 = 6;
        ArraysHelper.printArray(twoSum1(arr1, target1));


    }

    public static int[] twoSum(int[] nums, int target) {

        int len = nums.length;


        for (int start = 0; start < len - 1; start++) {

            for (int end = start + 1; end < len; end++) {

                if (start + end == target) {
                    return new int[]{start, end};
                }

            }


        }
        return new int[2];

    }


    public static int[] twoSum1(int[] nums, int target) {

        int len = nums.length;

        HashMap<Integer, Integer> map = new HashMap<Integer, Integer>();
        int[] result = new int[2];

        for (int i = 0; i < len; i++) {
            map.put(nums[i], i);
        }

        for (int i = 0; i < len; i++) {
            int first = nums[i];
            int second = target - first;
            Integer val = map.get(second);
            if (val != null && val != i) {
                result[0] = i;
                result[1] = val;
                break;
            }

        }

        return result;
    }


}

File: src/leetcode/array/twoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    
    let map = {}
    let arr = [];
    for (let i =0 ; i< nums.length; i++){
        //check if second part already in array
        if(map[target-nums[i]] != undefined){
            arr.push(map[target-nums[i]]);
            arr.push(i);
            break;
        }else{
        map[nums[i]] = i;
        }
    }
    return arr;
};


const nums = [0,7,8,2,5,3];
const result  =  twoSum(nums, 9);
console.log(result);